pandas/tests/series/methods/test_astype.py

```diff
@@ -470,6 +470,18 @@ def test_astype_categories_raises(self):
         with pytest.raises(TypeError, match="got an unexpected"):
             s.astype("category", categories=["a", "b"], ordered=True)
 
+    def test_astype_str_to_extension_dtype(self):
+        # GH-40351
+        s = Series(["A", np.NaN], dtype="string")
+        result = s.astype("category")
+        expected = Series(["A", np.NaN], dtype="category")
+        tm.assert_series_equal(result, expected)
+
+        s = Series(["1/1/2021", "2/1/2021"], dtype="string")
```

Comment:
can u add an example for Timedelta, Datetime w/time zone and Interval (all the EA types)

Reply:
I added test for all ExtensionArray dtypes. `BooleanArray` and `IntervalArray` I had to exclude since there's no way to to parse them back from a list of strings. `TimedeltaArray` xfails due to #40478. For `PeriodArray`and `DatetimeArray` the `NaT` get converted to `NA` strings. But converting the `NA` strings back to `NaT` fails. I added XFails, unless expecting successful EA ⇒ StringArray ⇒ EA roundtrips are still up for debate.